htlcswitch/switch.go

```diff
@@ -983,24 +987,19 @@ func (s *Switch) parseFailedPayment(deobfuscator ErrorDecrypter,
 	// A regular multi-hop payment error that we'll need to
 	// decrypt.
 	default:
-		var err error
 		// We'll attempt to fully decrypt the onion encrypted
 		// error. If we're unable to then we'll bail early.
-		failure, err = deobfuscator.DecryptError(htlc.Reason)
+		failure, err := deobfuscator.DecryptError(htlc.Reason)
 		if err != nil {
 			userErr := fmt.Sprintf("unable to de-obfuscate "+
 				"onion failure (hash=%v, pid=%d): %v",
 				paymentHash, paymentID, err)
 			log.Error(userErr)
-			failure = &ForwardingError{
-				FailureSourceIdx: 0,
-				ExtraMsg:         userErr,
-				FailureMessage:   lnwire.NewTemporaryChannelFailure(nil),
-			}
+			return ErrUnreadableFailureMessage
```

Comment:
nit: since `userErr` is no longer returned in the failure, unable should be capitalized.

Reply:
fixed